src/pool/srv_pool.c

```diff
@@ -255,6 +260,12 @@ pool_prop_default_copy(daos_prop_t *prop_def, daos_prop_t *prop)
 					return rc;
 			}
 			break;
+		case DAOS_PROP_PO_GLOBAL_VERSION:
+			D_ERROR("pool global version property could be not set\n");
+			return -DER_NO_PERM;
```

Comment:
I would lean against -DER_NO_PERM as the return code here. -DER_NO_PERM is used to represent access denied via the ACL. I'd suggest either using -DER_INVAL or adding a new DER code (-DER_IMMUTABLE or something like that?)